iree/turbine/kernel/wave/utils/general_utils.py

```diff
@@ -126,9 +126,13 @@ def remove_global_indexing(
     for key in new_index:
         for constraint in tiling_constraints:
             new_dim = new_index[key]
-            if sympy.sympify(new_dim.start).has(constraint.induction_var):
+            if new_dim.has(constraint.induction_var):
                 new_dim = new_dim.subs({constraint.induction_var: 0})
-                new_dim.start = new_dim.start - constraint.start
+                if isinstance(new_dim, IndexSequence):
+                    new_dim.start = new_dim.start - constraint.start
+                else:
+                    new_dim = new_dim - constraint.start
```

Comment:
this probably need another isinstance protection, and then we can have an else that raise ValueError